src/stackwalk.c

```diff
@@ -1249,6 +1249,47 @@ return 0;
 #endif
 }
 
+JL_DLLEXPORT size_t jl_try_record_thread_backtrace(jl_ptls_t ptls2, jl_bt_element_t *bt_data, size_t max_bt_size) JL_NOTSAFEPOINT
+{
+    int16_t tid = ptls2->tid;
+    jl_task_t *t = NULL;
+    bt_context_t *context = NULL;
+    bt_context_t c;
+    int lockret = jl_lock_stackwalk();
+    if (!jl_thread_suspend_and_get_state(tid, 1, &c)) {
+        jl_unlock_stackwalk(lockret);
+        return 0;
+    }
+    jl_unlock_stackwalk(lockret);
+    // thread is stopped, safe to read the task it was running before we stopped it
+    t = jl_atomic_load_relaxed(&ptls2->current_task);
+    if (context == NULL && (!t->ctx.copy_stack && t->ctx.started && t->ctx.ctx != NULL)) {
+        // need to read the context from the task stored state
+        jl_jmp_buf *mctx = &t->ctx.ctx->uc_mcontext;
+#if defined(_OS_WINDOWS_)
+        memset(&c, 0, sizeof(c));
+        if (jl_simulate_longjmp(*mctx, &c))
+            context = &c;
+#elif defined(JL_HAVE_UNW_CONTEXT)
+        context = t->ctx.ctx;
+#elif defined(JL_HAVE_UCONTEXT)
+        context = jl_to_bt_context(t->ctx.ctx);
+#elif defined(JL_HAVE_ASM)
+        memset(&c, 0, sizeof(c));
+        if (jl_simulate_longjmp(*mctx, &c))
+            context = &c;
+#else
+     #pragma message("jl_try_record_thread_backtrace not defined for unknown task system")
+#endif
+    }
+    if (context == NULL) {
+        return 0;
+    }
+    size_t bt_size = rec_backtrace_ctx(bt_data, max_bt_size, context, t->gcstack);
```

Comment:
`t->gcstack` is UB here because of how you stopped the thread
```suggestion
    size_t bt_size = rec_backtrace_ctx(bt_data, max_bt_size, context, NULL);
```

Reply:
Actually, no, I guess `rec_backtrace_ctx` only reads particular sorts of entries, and those entries are carefully designed to not be UB, so there are only benign races here (unless t->previous_task is set, indicating it is in the process of a task switch such that we don't know if we should use t->previous_task->gcstack or t->gcstack)
```suggestion
    size_t bt_size = rec_backtrace_ctx(bt_data, max_bt_size, context, t->previous_task ? NULL : t->gcstack);
```